koku/api/dataexport/syncer/__init__.py

```diff
@@ -94,40 +97,70 @@ def _copy_object(self, s3_destination_bucket, source_object):
+            log_uuid = str(uuid.uuid4())
+            LOG.info(
+                '%s Beginning sync_bucket to %s for %s from %s to %s',
+                log_uuid,
+                s3_destination_bucket_name,
+                schema_name,
+                date_range[0],
+                date_range[1],
+            )
             start_date, end_date = date_range
             # rrule is inclusive for both dates, so we need to make end_date exclusive
             end_date = end_date - timedelta(days=1)
             days = rrule(DAILY, dtstart=start_date, until=end_date)
             months = rrule(MONTHLY, dtstart=start_date, until=end_date)
             s3_destination_bucket = self.s3_resource.Bucket(s3_destination_bucket_name)
+            providers = Provider.objects.filter(customer__schema_name=schema_name).all()
 
             # Copy the specific month level files
-            for month, provider_type in product(months, _PROVIDER_TYPES):
+            for month, provider in product(months, providers):
+                # We need to normalize capitalization and "-local" dev providers.
+                provider_slug = provider.type.lower().split('-')[0]
+                prefix = (
+                    f'{settings.S3_BUCKET_PATH}/{schema_name}/'
+                    f'{provider_slug}/{provider.uuid}/'
+                    f'{month.year:04d}/{month.month:02d}/00/'
+                )
+                LOG.debug('sync_bucket checking prefix %s', prefix)
                 for source_object in self.s3_source_bucket.objects.filter(
-                    Prefix=(
-                        f'{settings.S3_BUCKET_PATH}/{account}/{provider_type}/'
-                        f'{month.year:04d}/{month.month:02d}/00/'
-                    )
+                    Prefix=prefix
                 ):
                     self._copy_object(s3_destination_bucket, source_object)
 
             # Copy all the day files
-            for day, provider_type in product(days, _PROVIDER_TYPES):
+            for day, provider in product(days, providers):
+                # We need to normalize capitalization and "-local" dev providers.
+                provider_slug = provider.type.lower().split('-')[0]
+                prefix = (
+                    f'{settings.S3_BUCKET_PATH}/{schema_name}/'
+                    f'{provider_slug}/{provider.uuid}/'
+                    f'{day.year:04d}/{day.month:02d}/{day.day:02d}/'
+                )
+                LOG.debug('sync_bucket checking prefix %s', prefix)
                 for source_object in self.s3_source_bucket.objects.filter(
-                    Prefix=(
-                        f'{settings.S3_BUCKET_PATH}/{account}/{provider_type}/'
-                        f'{day.year:04d}/{day.month:02d}/{day.day:02d}/'
-                    )
+                    Prefix=prefix
                 ):
                     self._copy_object(s3_destination_bucket, source_object)
+
+            log_uuid = str(uuid.uuid4())
```

Comment:
Is the point of this uuid to keep track of a single sync task?
If so shouldn't we be using the uuid declared above?

Reply:
Good catch! Fixing this in the next push.